Tools/clinic/clinic.py

```diff
@@ -6375,26 +6320,22 @@ def add_parameter(text: str) -> None:
         # therefore this is commented out:
         #
         # if f.return_converter.py_default:
-        #     add(' -> ')
-        #     add(f.return_converter.py_default)
+        #     acc.add(' -> ')
+        #     acc.add(f.return_converter.py_default)
 
         if not f.docstring_only:
-            add("\n" + sig_end_marker + "\n")
+            lines.append("\n" + sig_end_marker + "\n")
 
-        signature_line = output()
+        signature_line = "".join(lines)
 
         # now fix up the places where the brackets look wrong
         return signature_line.replace(', ]', ',] ')
 
     @staticmethod
     def format_docstring_parameters(params: list[Parameter]) -> str:
         """Create substitution text for {parameters}"""
-        add, output = text_accumulator()
-        for p in params:
-            if p.docstring:
-                add(p.render_docstring())
-                add('\n')
-        return output()
+        lines = [p.render_docstring() + "\n" for p in params if p.docstring]
+        return "".join(lines)
```

Comment:
```suggestion
        return "".join(p.render_docstring() + "\n" for p in params if p.docstring)
```

We could even just get rid of this method and inline the call: it's only called in one place.

Reply:
Yes, absolutely.

Reply:
OTOH, the method name increases the readability slightly.

Reply:
Sure, we can keep it as a separate method!